engine/api/src/main/java/io/deephaven/engine/table/Table.java

```diff
@@ -1763,56 +1762,57 @@ Table updateBy(@NotNull final UpdateByControl control,
-     * provided before applying the operation.
+     * Create a table with the same {@link #getRowSet() rowSet} as its parent that will perform the specified row based
+     * operations to it. As opposed to {@link #update(String...)} these operations are more restricted but are
+     * capable of processing state between rows. The operation will be applied to each group individually (identified
+     * by the set of key columns provided by the user).
      *
      * @param operations the operations to apply to the table.
      * @param byColumns the columns to group by before applying.
-     * @return a table with the same index, with the specified operations applied to each group defined by the
+     * @return a table with the same rowSet, with the specified operations applied to each group defined by the
      *         {@code byColumns}
      */
     @ConcurrentMethod
     Table updateBy(@NotNull final Collection<? extends UpdateByClause> operations, final String... byColumns);
 
     /**
-     * Create a table with the same {@link #getRowSet() rowSet} as its parent that will perform the specified set of row
-     * based operations to it. As opposed to {@link #update(String...)} these operations are more restricted but are
-     * capable of processing state between rows. This operation will group the table by the specified set of keys if
-     * provided before applying the operation.
+     * Create a table with the same {@link #getRowSet() rowSet} as its parent that will perform the specified row based
+     * operations to it. As opposed to {@link #update(String...)} these operations are more restricted but are
+     * capable of processing state between rows. The operation will be applied to each group individually (identified
+     * by the set of key columns provided by the user).
      *
      * @param operations the operations to apply to the table.
      * @param byColumns the columns to group by before applying.
-     * @return a table with the same index, with the specified operations applied to each group defined by the
+     * @return a table with the same rowSet, with the specified operations applied to each group defined by the
      *         {@code byColumns}
      */
     @ConcurrentMethod
     Table updateBy(@NotNull Collection<? extends UpdateByClause> operations,
             @NotNull Collection<? extends Selectable> byColumns);
 
     /**
-     * Create a table with the same {@link #getRowSet() rowSet} as its parent that will perform the specified set of row
-     * based operations to it. As opposed to {@link #update(String...)} these operations are more restricted but are
-     * capable of processing state between rows. This operation will group the table by the specified set of keys if
-     * provided before applying the operation.
+     * Create a table with the same {@link #getRowSet() rowSet} as its parent that will perform the specified row based
```

Comment:
I agree that this language is hard to get right. Proposed the following over Slack:
>Create a table with the same  {@link #getRowSet() row set} as its parent, with additional columns added according to the specified {@code operations}. Operations apply position or time-based windowing functions in order to determine the set of responsive rows from a given input row’s group (as determined by the {@code byColumns}), and then compute the output column value(s) as an incremental aggregation over the responsive rows. In contrast to update or select, updateBy has no correctness concerns for multi-row computations.